dev-support/code-formatter/hadoop_idea_formatter.xml

```diff
@@ -0,0 +1,76 @@
+<!--
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<code_scheme name="Hadoop" version="173">
+  <option name="RIGHT_MARGIN" value="80" />
+  <JavaCodeStyleSettings>
+    <option name="CLASS_COUNT_TO_USE_IMPORT_ON_DEMAND" value="999" />
+    <option name="NAMES_COUNT_TO_USE_IMPORT_ON_DEMAND" value="999" />
+    <option name="PACKAGES_TO_USE_IMPORT_ON_DEMAND">
+      <value />
+    </option>
+    <option name="IMPORT_LAYOUT_TABLE">
+      <value>
+        <package name="org.apache.hadoop.thirdparty" withSubpackages="true" static="false" />
+        <emptyLine />
+        <package name="" withSubpackages="true" static="false" />
+        <emptyLine />
+        <package name="javax" withSubpackages="true" static="false" />
+        <package name="java" withSubpackages="true" static="false" />
+        <emptyLine />
+        <package name="" withSubpackages="true" static="true" />
+      </value>
+    </option>
```

Comment:
This takes care of import ordering:

- import org.apache.hadoop.thirdparty.*
- `<br />`
- import everything non-static in order
- `<br />`
- import javax.*
- import java.*
- `<br />`
- import everything static

Reply:
that's not our ordering though, is it?

Reply:
I tried to confirm from some of the oldest classes e.g Namenode and few others (within namenode package). But now I see that some other similar old classes have different order. For instance, Datanode itself has kind of reverse order.
@steveloughran do we have the correct order defined somewhere?

Reply:
I guess this is the correct order https://github.com/apache/hadoop/pull/2073#pullrequestreview-434899055

in addition just thirdparty imports goes to the other block only.

Do wait for Steve to confirm once before updating

Reply:
> I guess this is the correct order [#2073 (review)](https://github.com/apache/hadoop/pull/2073#pullrequestreview-434899055)
> 
> in addition just thirdparty imports goes to the other block only.
> 
> Do wait for Steve to confirm once before updating

I see, Thanks. Unless we follow module specific formatting (few rules are applicable to S3A, ABFS, WASB and the like only, hence wondering), I think this should be good enough to keep in formatter as generic rule.